controllers/integration/integration_adapter.go

```diff
@@ -137,6 +137,139 @@ func (a *Adapter) getAllApplicationSnapshots() (*[]appstudioshared.ApplicationSn
 	return &applicationSnapshots.Items, nil
 }
 
+func (a *Adapter) getAllEnvironments() (*[]appstudioshared.Environment, error) {
+
+	environmentList := &appstudioshared.EnvironmentList{}
+	opts := []client.ListOption{
+		client.InNamespace(a.application.Namespace),
+	}
+	err := a.client.List(a.context, environmentList, opts...)
+	return &environmentList.Items, err
+}
+
+func (a *Adapter) findAvailabeEnvironments() (*[]appstudioshared.Environment, error) {
+	allEnvironments, err := a.getAllEnvironments()
+	if err != nil {
+		return nil, err
+	}
+	availableEnvironments := []appstudioshared.Environment{}
+	for _, environment := range *allEnvironments {
+		if environment.Spec.ParentEnvironment == "" {
+			availableEnvironments = append(availableEnvironments, environment)
+		} else {
+			for _, tag := range environment.Spec.Tags {
+				if tag == "ephemeral" {
```

Comment:
In the description of https://issues.redhat.com/browse/HACBS-801, it is said `Exclude all "ephemeral" environments created by integration controller from this set`, so we should exclude environment with tag `ephemeral`

Reply:
I just notice this PR is on your own repo, I thought I didn't commit on the redhat-appstudio successful. So please ignore comment here.